src/libraries/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -395,28 +396,28 @@ private bool AgeCookies(string domain)
             }
             lock (m_domainTable.SyncRoot)
             {
-                foreach (DictionaryEntry entry in m_domainTable)
+                foreach (DictionaryEntry? entry in m_domainTable)
                 {
                     if (domain == null)
                     {
-                        tempDomain = (string)entry.Key;
-                        pathList = (PathList)entry.Value; // Aliasing to trick foreach
+                        tempDomain = (string?)entry?.Key;
+                        pathList = (PathList)entry?.Value!; // Aliasing to trick foreach
                     }
                     else
                     {
                         tempDomain = domain;
-                        pathList = (PathList)m_domainTable[domain];
+                        pathList = (PathList)m_domainTable[domain]!;
                     }
 
                     domain_count = 0; // Cookies in the domain
                     lock (pathList.SyncRoot)
                     {
-                        foreach (CookieCollection cc in pathList.Values)
+                        foreach (CookieCollection? cc in pathList.Values)
                         {
-                            itemp = ExpireCollection(cc);
+                            itemp = ExpireCollection(cc!);
```

Comment:
If the collection won't be null, we can Debug.Assert that at the beginning of the loop rather than ! on each access.

Reply:
Added Debug.Assert().